src/Features/Core/Portable/Completion/CompletionOptionsMetadata.cs

```diff
@@ -0,0 +1,109 @@
+                storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.SnippetsBehavior"));
+
+        public static readonly PerLanguageOption2<bool> ShowNameSuggestions =
+            new(nameof(CompletionOptions), nameof(ShowNameSuggestions), CompletionOptions.Default.ShowNameSuggestions,
+            storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.ShowNameSuggestions"));
+
+        //Dev16 options
+
+        // Use tri-value so the default state can be used to turn on the feature with experimentation service.
+        public static readonly PerLanguageOption2<bool?> ShowItemsFromUnimportedNamespaces =
+            new(nameof(CompletionOptions), nameof(ShowItemsFromUnimportedNamespaces), CompletionOptions.Default.ShowItemsFromUnimportedNamespaces,
+            storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.ShowItemsFromUnimportedNamespaces"));
+
+        public static readonly PerLanguageOption2<bool> TriggerInArgumentLists =
+            new(nameof(CompletionOptions), nameof(TriggerInArgumentLists), CompletionOptions.Default.TriggerInArgumentLists,
+            storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.TriggerInArgumentLists"));
+
+        /// <summary>
+        /// Indicates if the completion is trigger by toggle the expander.
+        /// </summary>
+        public static readonly Option2<bool> IsExpandedCompletion
+            = new("CompletionServiceOptions", nameof(IsExpandedCompletion), CompletionOptions.Default.IsExpandedCompletion);
+
+        /// <summary>
+        /// Timeout value used for time-boxing completion of unimported extension methods.
+        /// Value less than 0 means no timebox; value == 0 means immediate timeout (for testing purpose)
+        /// </summary>
+        public static readonly Option2<int> TimeoutInMillisecondsForExtensionMethodImportCompletion
+            = new("CompletionServiceOptions", nameof(TimeoutInMillisecondsForExtensionMethodImportCompletion), CompletionOptions.Default.TimeoutInMillisecondsForExtensionMethodImportCompletion);
+
+        // Embedded languages:
+
+        public static PerLanguageOption2<bool> ProvideRegexCompletions =
+            new(
+                "RegularExpressionsOptions",
+                nameof(ProvideRegexCompletions),
+                CompletionOptions.Default.ProvideRegexCompletions,
+                storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.ProvideRegexCompletions"));
+
+        public static readonly PerLanguageOption2<bool> ProvideDateAndTimeCompletions =
+            new(
+                "DateAndTime",
+                nameof(ProvideDateAndTimeCompletions),
+                CompletionOptions.Default.ProvideDateAndTimeCompletions,
+                storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.ProvideDateAndTimeCompletions"));
+    }
```

Comment:
is there no need for an options provider anymore?  (is that one of the points of hte change?  these don't get mef'ed in anymore.  instead the client code instead uses the methods/extensions to get these values from the IGlobalOptionService and then uses that to create an instance of CompletionOptions?

Reply:
The only place where IOptionProvider is currently used is for solution options serialization. For global options its not used.
Options participate in storage via IOptionPersister.